stacker_blueprints/iam_roles.py

```diff
@@ -36,21 +52,60 @@ def __init__(self, *args, **kwargs):
 
     def create_role(self, name, assumerole_policy):
         t = self.template
+        v = self.get_variables()
+
+        role_kwargs = {
+            'AssumeRolePolicyDocument': assumerole_policy,
+        }
+
+        attached_policies = v['AttachedPolicies']
+        if attached_policies and len(attached_policies):
+            role_kwargs['ManagedPolicyArns'] = attached_policies
+
+        path = v['Path']
+        if path:
+            role_kwargs['Path'] = path
 
         role = t.add_resource(
             iam.Role(
                 name,
-                AssumeRolePolicyDocument=assumerole_policy,
+                **role_kwargs
             )
         )
 
         t.add_output(
             Output(name + "RoleName", Value=Ref(role))
         )
+
         t.add_output(
             Output(name + "RoleArn", Value=GetAtt(role.title, "Arn"))
         )
 
+        if v['InstanceProfile']:
```

Comment:
This won't work the way you want, I don't believe, since it's creating an `InstanceProfile` PER `Role` you create, but it's naming the `InstanceProfile` output the same each time.

Is the intent to create an instance profile per role, or one instance profile with all of the roles?

Also, is it worth checking that the role is an ec2 role, and not a Lambda role, before creating the profile?